### DISCUSSION TITLE
expiration date?

### DISCUSSION THREAD

**Discussion by Curtmcgirt** [Venafi/venafips#103] [Q&A] [answered]
i see how i can use the three 'expire' switches on find-venaficertificate to get certificates that match criteria, but what command(s) do i use to return the actual expiration dates in results? i don't see ExpirationDate in the output of get-tppattribute either.

**Comment by gdbarron** [comment, marked as answer]
Find-VenafiCertificate searches and gets basic info. Detailed info can be found with Get-VenafiCertificate.

**Reply by gdbarron** [reply]
Not sure command you are running, but Get-TppAttribute should work as well.

**Comment by Curtmcgirt** [comment]
get-tppattribute $cert 
 does not return an 'expires' or 'validto' attribute. i also don't see anything promising with

Get-TppClassAttribute -ClassName 'X509 Server Certificate'

i was going to try to do some get-date math with the last renewed date when i saw a "validity period" attribute, but 

Get-TppAttribute -path $cert -Attribute 'Validity Period' says the 'value' is null.

**Reply by Curtmcgirt** [reply]
this does work:. 

Get-VenafiCertificate -path $cert | select -ExpandProperty certificatedetails | select -ExpandProperty validto

for some reason i was thinking  the 'get' command downloaded the certificate.

**Reply by gdbarron** [reply]
Looks like valid to isn't an attribute after all.  This does make sense as the attribute is not of the certificate per se, but of the tpp object.

'Get' retrieves certificate details whereas 'Export' downloads the certificate.
